yarn-project/sequencer-client/src/publisher/l1-publisher.test.ts

```diff
@@ -40,45 +56,84 @@ describe('L1Publisher', () => {
     const result = await publisher.processL2Block(l2Block);
 
     expect(result).toEqual(true);
-    expect(txSender.sendProcessTx).toHaveBeenCalledWith({ header, archive, body, proof });
-    expect(txSender.getTransactionReceipt).toHaveBeenCalledWith(txHash);
+    expect(txSender.sendProcessTx).toHaveBeenCalledWith({ header, archive, txsHash, body, proof });
+    expect(txSender.getTransactionReceipt).toHaveBeenCalledWith(processTxHash);
   });
 
   it('does not publish if last archive root is different to expected', async () => {
     txSender.getCurrentArchive.mockResolvedValueOnce(L2Block.random(43).archive.root.toBuffer());
     const result = await publisher.processL2Block(l2Block);
     expect(result).toBe(false);
+    expect(txSender.sendPublishTx).not.toHaveBeenCalled();
     expect(txSender.sendProcessTx).not.toHaveBeenCalled();
   });
 
-  it('does not retry if sending a tx fails', async () => {
-    txSender.sendProcessTx.mockReset().mockRejectedValueOnce(new Error()).mockResolvedValueOnce(txHash);
+  it('does not retry if sending a publish tx fails', async () => {
+    txSender.sendPublishTx.mockReset().mockRejectedValueOnce(new Error()).mockResolvedValueOnce(processTxHash);
```

Comment:
Again, should this not be the `publishTxHash`?

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/4118/commits/19b1eeb879754943d589f4f0174999cf4012c47a